iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/OperatorTreeGenerator.java

```diff
@@ -3348,4 +3358,126 @@ public Operator visitExplainAnalyze(ExplainAnalyzeNode node, LocalExecutionPlanC
+    dataDriverContext.setQueryDataSourceType(QueryDataSourceType.DEVICE_REGION_SCAN);
+
+    return regionScanOperator;
+  }
+
+  @Override
+  public Operator visitTimeSeriesRegionScan(
+      TimeseriesRegionScanNode node, LocalExecutionPlanContext context) {
+    OperatorContext operatorContext =
+        context
+            .getDriverContext()
+            .addOperatorContext(
+                context.getNextOperatorId(),
+                node.getPlanNodeId(),
+                ActiveTimeSeriesRegionScanOperator.class.getSimpleName());
+    Filter filter = context.getGlobalTimeFilter();
+    DataDriverContext dataDriverContext = (DataDriverContext) context.getDriverContext();
+
+    Map<IDeviceID, Map<String, TimeseriesSchemaInfo>> timeseriesToSchemaInfo = new HashMap<>();
+    for (Map.Entry<PartialPath, Map<PartialPath, List<TimeseriesSchemaInfo>>> entryMap :
+        node.getDeviceToTimeseriesSchemaInfo().entrySet()) {
+      Map<String, TimeseriesSchemaInfo> timeseriesSchemaInfoMap =
+          getTimeseriesSchemaInfoMap(entryMap, dataDriverContext);
+      timeseriesToSchemaInfo.put(
+          new PlainDeviceID(entryMap.getKey().getFullPath()), timeseriesSchemaInfoMap);
+    }
+    ActiveTimeSeriesRegionScanOperator regionScanOperator =
+        new ActiveTimeSeriesRegionScanOperator(
+            operatorContext, node.getPlanNodeId(), timeseriesToSchemaInfo, filter);
+
+    dataDriverContext.addSourceOperator(regionScanOperator);
+    dataDriverContext.setQueryDataSourceType(QueryDataSourceType.TIME_SERIES_REGION_SCAN);
+
+    return regionScanOperator;
+  }
+
+  private static Map<String, TimeseriesSchemaInfo> getTimeseriesSchemaInfoMap(
+      Map.Entry<PartialPath, Map<PartialPath, List<TimeseriesSchemaInfo>>> entryMap,
+      DataDriverContext context) {
+    Map<String, TimeseriesSchemaInfo> timeseriesSchemaInfoMap = new HashMap<>();
+    for (Map.Entry<PartialPath, List<TimeseriesSchemaInfo>> entry :
+        entryMap.getValue().entrySet()) {
+      PartialPath path = entry.getKey();
+      context.addPath(path);
+      if (path instanceof MeasurementPath) {
+        TimeseriesSchemaInfo timeseriesSchemaInfo = entry.getValue().get(0);
+        timeseriesSchemaInfo.setDataBase(path.getDevicePath().getDevice());
+        timeseriesSchemaInfoMap.put(path.getMeasurement(), timeseriesSchemaInfo);
+      } else if (path instanceof AlignedPath) {
+        AlignedPath alignedPath = (AlignedPath) path;
+        List<String> measurementList = alignedPath.getMeasurementList();
+        if (measurementList.size() != entry.getValue().size()) {
+          throw new IllegalArgumentException(
+              "The size of measurementList and timeseriesSchemaInfoList should be equal in aligned path.");
+        }
+        for (int i = 0; i < measurementList.size(); i++) {
+          TimeseriesSchemaInfo timeseriesSchemaInfo = entry.getValue().get(i);
+          timeseriesSchemaInfo.setDataBase(alignedPath.getDevicePath().getDevice());
```

Comment:
database is not the previous level of device Path, what's the database name of root.db.a..b.c.d1.s1?